algebird-core/src/main/scala/com/twitter/algebird/macros/GroupMacro.scala

```diff
@@ -0,0 +1,45 @@
+package com.twitter.algebird.macros
+
+import scala.language.experimental.{ macros => sMacros }
+import scala.reflect.macros.Context
+import scala.reflect.runtime.universe._
+
+import com.twitter.algebird._
+
+object GroupMacro {
+  def caseClassGroup[T](c: Context)(implicit T: c.WeakTypeTag[T]): c.Expr[Group[T]] = {
+    import c.universe._
+
+    macros.ensureCaseClass(c)
+
+    val res = q"""
+    new _root_.com.twitter.algebird.Group[$T] {
+      ${SemigroupMacro.plus(c)}
```

Comment:
this is a really nice solution, and the code reuse is great. There is one issue here: what if you have conflicting `Semigroup[T]` and `Group[T]`. For instance, `Semigroup[Int]` could be multiplication, so `plus` would actually be addition, but `Group[Int]` is going to have the `plus` method be addition.

This won't happen by default, but a user could shoot themselves in the foot.

We could avoid this by getting all the implicit `Group`s for the member types, and then explicitly passing those into the SemigroupMacro so it does not resolve them again. It's not immediately clear the best way to do that. Can you try something like that?